### PR TITLE
chore(deps): update js dev dependencies

### PR DESCRIPTION
**Pull request by Legend-Master**
`eslint-plugin-import`, `eslint-plugin-n`, `eslint-plugin-node`, `eslint-plugin-promise` were unused before, and are removed  in this PR

### CODE REVIEW

**Review by FabianLars — DISMISSED** [review]

File: tooling/api/package.json [hunk 1]
```diff
@@ -39,33 +39,31 @@
     "build": "rollup -c --configPlugin typescript",
     "npm-pack": "yarn build && cd ./dist && npm pack",
     "npm-publish": "yarn build && cd ./dist && yarn publish --access public --loglevel silly --tag next",
-    "ts:check": "tsc -noEmit",
-    "lint": "eslint --ext ts \"./src/**/*.ts\"",
-    "lint:fix": "eslint --fix --ext ts \"./src/**/*.ts\"",
+    "ts:check": "tsc",
```
Comment (amrbashir) [inline tooling/api/package.json]: any reason to remove `-noEmit`?

Comment (Legend-Master) [reply tooling/api/package.json]: We already have `"noEmit": true` in `tsconfig.json`

Comment (amrbashir) [reply tooling/api/package.json]: hmmm, seems better to leave it here as well, just to not be confused.

Comment (Legend-Master) [reply tooling/api/package.json]: I have no preference on this, if you prefer that, I'll bring it back

**Review by amrbashir — APPROVED** [review]

**Comment by amrbashir** [comment]
Thank  you

**Comment by amrbashir** [comment]
clippy errors are fixed in https://github.com/tauri-apps/tauri/pull/10054

**Pull request merged**